.github/workflows/generate.yaml

```
@@ -0,0 +1,24 @@
+
+# We use github actions to test the code on windows and linux amd64. Circleci is used for linux arm64.
+
+name: generate-mock
+on:
+  pull_request:
+
+jobs:
+  native:
+    strategy:
+      matrix:
+        runs-on: [ ubuntu-latest, windows-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - run: go install github.com/golang/mock/mockgen@v1.6.0
```

I don't think you need to do this since `go generate` is a module-aware command? https://go.dev/ref/mod#build-commands

(and I think removing it would remove the need to update the version pinned here when the version pinned in `go.mod` is updated)

---

I was confused here: we still need to install the package. But we should avoid pinning it here as well

As a conclusion to our a convo on slack, let's replace this with:

```suggestion
      - run: go install github.com/golang/mock/mockgen
```
so that we install the package at the version defined in `go.mod`.